src/Rings/MPolyMap/flattenings.jl

```diff
@@ -468,12 +468,9 @@ function flatten(R::MPolyQuoRing; cached::Bool=false)
 end
 
 function (phi::RingFlattening)(I::MPolyIdeal)
-  if haskey(flat_counterparts(phi), I)
-    return flat_counterparts(phi)[I][1]
-  end
-  I_flat = ideal(codomain(phi), elem_type(codomain(phi))[phi(g) for g in gens(I)])
-  flat_counterparts(phi)[I] = (I_flat, phi)
-  return I_flat
+  return get!(flat_counterparts(phi), I) do
+    return ideal(codomain(phi), elem_type(codomain(phi))[phi(g) for g in gens(I)])
+  end::Any  # TODO: add better type annotation
```

Comment:
Presumably this could be improved using `ideal_type` but I couldn't be bothered to figure out the details. Something like
```suggestion
  end::ideal_type(SOMETHING)
```
Or perhaps the returned ideal will always have the same type as the input ideal? Then we could use
```suggestion
  end::typeof(I)
```

Perhaps @HechtiDerLachs knows?

Reply:
Unfortunately, the second will probably not work. As for the first: We did not introduce such a getter. It would be part of an officially declared ideal interface. But we do not have that, yet. So the TODO has to stay for the moment.

Reply:
Actually I added `ideal_type` to AA some time ago :-) but of course there is no good way you could have known that -- another reason we need a well-curated CHANGELOG.md and perhaps release announcements